### ISSUE TITLE
pretty printing expint, polylog etc with opportunistic subscripts

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#25312]
Currently, we have:
```
>>> init_printing()
>>> expint(3,z)     # I look very nice!
E₃(z)

>>> expint(s, z)    # I could look nicer!
expint(s, z)

>>> expint(cos(s - 1)/17, z)    # but leave me be
      ⎛cos(s - 1)   ⎞
expint⎜──────────, z⎟
      ⎝    17       ⎠
```
`polylog` will behave the same after #25311.  Both use
``` 
       if e.args[0].is_Integer and self._use_unicode:
```

Perhaps one could use a helper function based on the list of things which can be subscripted:
```
from sympy.printing.pretty.pretty_symbology import sub

{'a': 'ₐ',
 'e': 'ₑ',
 'i': 'ᵢ',
 'o': 'ₒ',
 'r': 'ᵣ',
 'u': 'ᵤ',
 'v': 'ᵥ',
 'x': 'ₓ',
 'h': 'ₕ',
 'k': 'ₖ',
 'l': 'ₗ',
 'm': 'ₘ',
 'n': 'ₙ',
 'p': 'ₚ',
 's': 'ₛ',
 't': 'ₜ',
 'beta': 'ᵦ',
 'gamma': 'ᵧ',
 'rho': 'ᵨ',
 'phi': 'ᵩ',
 'chi': 'ᵪ',
 '0': '₀',
 '1': '₁',
 '2': '₂',
 '3': '₃',
 '4': '₄',
 '5': '₅',
 '6': '₆',
 '7': '₇',
 '8': '₈',
 '9': '₉',
 '+': '₊',
 '-': '₋',
 '=': '₌',
 '(': '₍',
 ')': '₎'}
```

Maybe there already is such a helper function?  If not, I suggest writing an `opportunistic_subscripter` or some such helper function.

Bessel and Hankel could also potentially benefit from this printing.

**Comment by cbm755** [comment]
To be determined: `polylog(s - 1, z)` which at least in principle could be `Liₛ₋₁(z)` but could also be left as `polylog(s - 1, z)`

**Comment by bsach64** [comment]
Hello, I would like to help fix this issue.
As I understand it, we have to write an helper function (based on the list provided) to determine whether to pretty print the (expint) expression or not?

**Comment by cbm755** [comment]
Yes I think so.   The only other goal would be to have *both* `polylog` and `expint`  using the helper function.

I'm not an expert on pretty-printing but ping me on any MR you make and I'll see if I can help review.

**Comment by bsach64** [comment]
Will definitely do! Thanks!

**Comment by bsach64** [comment]
Hey @cbm755, I have made a draft pull request. I have tried my best to fix the problem and follow contributing guidelines. Please review the changes whenever you can and let me know what more to do! Thank you for your time!
Also,  `polylog(s - 1, z)` has been left as `polylog(s - 1, z)`
